access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java

```diff
@@ -163,7 +163,7 @@ public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
 
                     @Override
                     public MethodVisitor visitMethod(int access, final String methodName, final String methodDesc, String signature, String[] exceptions) {
-                        return new EmptyVisitor() {
+                        return new MethodVisitor(Opcodes.ASM5) {
```

Comment:
Actually I wasn't looking at the right line, the `MethodVisitor` at line 246 implements `visitMethodInsn`, not the one at line 166. That said, it's still not clear why it breaks. In ASM 5.2 `visitMethodInsn(int, String, String String)` calls `visitMethodInsn(int, String, String String, boolean)` when the api level used is `Opcodes.ASM5`. So even if the deprecated `visitMethodInsn` method is used it should still work.